### ISSUE TITLE
Bug: Game crashes while fighting Mutated Aechor Plant

### ISSUE THREAD

**Issue by 2468cool** [Builderdog841/Ancient-Aether#199]
I keep crashing while fighting the mutated aechor plant. I'm using the resistor shield and ray gun from alex's caves mod to fight the plant. This didn't happen the first time I was fighting the aechor plant even with the same weapons. There is no crash log I can post besides this screenshot. I hope this helps.
![crash](https://github.com/Builderdog841/Ancient-Aether/assets/83475290/d673e838-5ab9-4a98-845d-e21adacaeb27)

**Comment by 2468cool** [comment]
Now I was just using an iron axe and it crashed. It crashed the moment I got the green heart effect

**Comment by 2468cool** [comment]
It also crashes with that error without that affect. I'm not sure what is causing it.

**Comment by 2468cool** [comment]
After a quick test this bug does NOT occur in singleplayer

**Comment by Builderdog841** [comment]
That is a bit odd, I unfortunately don't have a proper way to test multiplayer bugs but I'll look into it once I have time

**Comment by 2468cool** [comment]
New error thrown. I saw this pop up once before but I didn't capture it at the time. Maybe this will help?
![crash2](https://github.com/Builderdog841/Ancient-Aether/assets/83475290/171aa691-4e37-4040-8aa3-3d5cf6127164)

**Comment by vnsmnn** [comment]
Hi after trying to find reason for my server crash I can confirm that I'm having the same issue when fighting Mutated Aechor Plant. The game crash when the plant fire the box green thingy 
![image](https://github.com/user-attachments/assets/dea9f4a8-20c0-4ddd-b413-d8ab49215df6)

**Comment by vnsmnn** [comment]
[2024-08-06-7.log.gz](https://github.com/user-attachments/files/16503678/2024-08-06-7.log.gz)
here is my client logs

**Comment by Capto7** [comment]
Yeah this still seems to be an issue as I am experiencing the same thing

**Comment by Capto7** [comment]
I found out why it happens! It's the projectiles it shoots out and then the effect it gives! When the effect wears off the first time, you are stuck at that place, continously getting kicked or crashing until you either defeat the boss or teleport away!

**Comment by VoidLeech** [comment]
Looking into this, no guarantees I'll figure it out, but:
f_132289_ is the `particle` field in ClientboundLevelParticlesPacket, where it's crashing in the write method due to attempting to get the type of a null particle.

**Comment by Capto7** [comment]
I really hope this will get fixed soon

**Comment by VoidLeech** [comment]
Reproduced in dev env, server logs:
https://mclo.gs/KHbljQ1

Root of the null particle is getting lost in transition, adding a mixin to ServerLevel so I can walk the stack and find it.

**Comment by VoidLeech** [comment]
Aaaaand there it is:
https://github.com/Builderdog841/Ancient-Aether/blob/bfcea9fc7548de401c480319732ced4564f4c3e9/src/main/java/net/builderdog/ancient_aether/entity/projectile/RemedyCrystal.java#L63-L65

**Comment by Laozhu132** [comment]
Crash again when I fight Mutated Aechor Plant using netherite sword in survival mode.
mod version:v0.9.9

![Image](https://github.com/user-attachments/assets/5fe4fa6c-6998-461b-9d26-12689c44862f)

**Comment by Builderdog841** [comment]
pr has been merged
